examples/optimize_coupling_scheme_parameter/search_toolbox.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Nov 22 14:15:08 2021
+
+Find optimal coupling scheme parameter for the staggered scheme,
+currently this parametrized scheme is only available for the HM-process. 
+
+@author: dominik kern
+"""
+
+from ogs6py.ogs import OGS
+from ogs6py.log_parser.log_parser import parse_file
+from ogs6py.log_parser.common_ogs_analyses import  fill_ogs_context, analysis_convergence_coupling_iteration, analysis_simulation_termination
+from scipy.optimize import minimize_scalar
+import numpy as np
+import pandas as pd
+
+
+class coupling_parameter_search:
+
+    def __init__(self, basename, ogs_dir=None):
```

Comment:
I think you should add doc strings based on the style used in ogs.py such that it can be interpreted by sphinx.